torch/csrc/jit/interned_strings.cpp

```diff
@@ -133,68 +95,33 @@ Symbol Symbol::fromQualString(const std::string & s) {
   return globalStrings().symbol(s);
 }
 
-const char* symbolNamespaceString(SymbolNamespace ns) {
-  switch (ns) {
-    case SymbolNamespace::onnx: return "onnx";
-    case SymbolNamespace::prim: return "prim";
-    case SymbolNamespace::aten: return "aten";
-    case SymbolNamespace::attr: return "attr";
-    case SymbolNamespace::scope: return "scope";
-    // NB: throwing an exception here causes gcc -O3 to produce far worse code
-    default: return "";
-  }
-}
-
-// NB: I really wanted to define this as std::strlen(symbolNamespaceString(ns)),
-// but gcc -O3 doesn't seem to be smart enough to push the std::strlen into
-// the switch statement.
-size_t symbolNamespaceLength(SymbolNamespace ns) {
-  switch (ns) {
-    case SymbolNamespace::onnx: return 4;
-    case SymbolNamespace::prim: return 4;
-    case SymbolNamespace::aten: return 4;
-    case SymbolNamespace::attr: return 4;
-    case SymbolNamespace::scope: return 5;
-    default: return 0;
-  }
-}
-
 const char * Symbol::toUnqualString() const {
-  return globalStrings().string(*this) + symbolNamespaceLength(ns()) + 2 /* double colon */;
+  return toQualString() + (globalStrings().string(ns()).second - strlen("namespaces"));
```

Comment:
What's going on here?

Reply:
I changed this to be more clear. We now just store a struct for each symbol:

```
  struct SymbolInfo {
    Symbol ns;
    std::string qual_name;
    std::string unqual_name;
  };
```

So there is no need for doing string math to get the unqual string.